lib/puma/launcher.rb

```diff
@@ -296,11 +296,7 @@ def prune_bundler
       deps, dirs = dependencies_and_files_to_require_after_prune
 
       log '* Pruning Bundler environment'
-      home = ENV['GEM_HOME']
```

Comment:
I thought original_env only looked at `BUNDLE_` environment variables.

Reply:
Do we have any test for this? Or can we write it?

Reply:
https://github.com/puma/puma/issues/653 looks like no test exists

Reply:
I'll try to implement tests for this, but if somebody can help me — I appreciate it, I'm not too familiar with Puma sources and environments.

Reply:
Also there are no tests for #1893. And I've found `prune_bundler` in `test/` directory only in `test/config/prune_bundler_with_deps.rb`, and I see no usage of this file in `test/`.

I've added `p ENV['GEM_HOME']` inside `Launcher#prune_bundler` and there is no output from tests.

Reply:
Oh, I've found `test/test_preserve_bundler_env.rb`, OK.

Reply:
You won't see that output because the Puma tests create new processes whose output is not piped to STDOUT. Take a look at the helpers/integration test code for more info.

The test for #1893 is here: https://github.com/puma/puma/pull/1893/files#diff-1a1a7e313ee72a0ca0e8eb22a2524879

Reply:
> Take a look at the helpers/integration test code for more info.

I've found `:log` option for `wait_for_server_to_boot`, yes, thank you.

Current results:

* `GEM_HOME` was `/home/alex/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0` and becomes `nil`
* `BUNDLE_GEMFILE` was `/home/alex/Projects/ruby/puma/test/bundle_preservation_test/Gemfile.bundle_env_preservation_test` and becomes `Gemfile.bundle_env_preservation_test`.

I'll try to add correct test for `GEM_HOME`. I think the change of `BUNDLE_GEMFILE` is not critical (tests passing).